napalm_nxos_ssh/nxos_ssh.py

```diff
@@ -0,0 +1,1513 @@
+        command = 'show running-config'
+        output = self.device.send_command(command)
+        section_username_tabled_output = napalm_base.helpers.textfsm_extractor(
+            self, 'users', output)
+
+        for user in section_username_tabled_output:
+            username = user.get('username', '')
+            if not username:
+                continue
+            if username not in users:
+                users[username] = _DEFAULT_USER_DICT.copy()
+
+            password = user.get('password', '')
+            if password:
+                users[username]['password'] = py23_compat.text_type(password.strip())
+
+            level = 0
+            role = user.get('role', '')
+            if role.startswith('priv'):
+                level = int(role.split('-')[-1])
+            else:
+                level = _CISCO_TO_CISCO_MAP.get(role, 0)
+            if level > users.get(username).get('level'):
+                # unfortunately on Cisco you can set different priv levels for the same user
+                # Good news though: the device will consider the highest level
+                users[username]['level'] = level
+
+            sshkeytype = user.get('sshkeytype', '')
+            sshkeyvalue = user.get('sshkeyvalue', '')
+            if sshkeytype and sshkeyvalue:
+                if sshkeytype not in ['ssh-rsa', 'ssh-dsa']:
+                    continue
+                users[username]['sshkeys'].append(py23_compat.text_type(sshkeyvalue))
+        return users
+
+    def traceroute(self,
+                   destination,
+                   source=c.TRACEROUTE_SOURCE,
+                   ttl=c.TRACEROUTE_TTL,
+                   timeout=c.TRACEROUTE_TIMEOUT,
+                   vrf=c.TRACEROUTE_VRF):
+
+        _HOP_ENTRY_PROBE = [
+            '\s+',
+            '(',  # beginning of host_name (ip_address) RTT group
+            '(',  # beginning of host_name (ip_address) group only
+            '([a-zA-Z0-9\.:-]*)',  # hostname
+            '\s+',
+            '\(?([a-fA-F0-9\.:][^\)]*)\)?'  # IP Address between brackets
+            ')?',  # end of host_name (ip_address) group only
+            # also hostname/ip are optional -- they can or cannot be specified
+            # if not specified, means the current probe followed the same path as the previous
+            '\s+',
+            '(\d+\.\d+)\s+ms',  # RTT
+            '|\*',  # OR *, when non responsive hop
+            ')'  # end of host_name (ip_address) RTT group
+        ]
+
+        _HOP_ENTRY = [
+            '\s?',  # space before hop index?
+            '(\d+)',  # hop index
+        ]
+
+        traceroute_result = {}
+        timeout = 5  # seconds
+        probes = 3  # 3 probes/jop and this cannot be changed on NXOS!
+
+        version = ''
+        try:
+            version = '6' if IPAddress(destination).version == 6 else ''
+        except AddrFormatError:
+            return {'error': 'Destination does not look like a valid IP Address: {}'.format(
```

Comment:
Do we not want possible DNS resolution?  I know it's allowed in the EOS driver

Reply:
Yes, we should fix that. Should fix it in NX-API and/or Cisco IOS as this was probably copied from one of those two places (probably from nx-api code).